examples/ex31p_proposed.cpp

```diff
@@ -0,0 +1,530 @@
+
+//                       MFEM Example 31 - Parallel Version
+//
+// Compile with: make ex31p
+//
+// Sample runs:  mpirun -np 4 ex31p -m ../data/hexagon.mesh -o 2
```

Comment:
The eigenmodes are very pretty :star-struck:, however, they do not appear to preserve the symmetries of the domain. Could this be a visualization issue or a convergence issue?  Or is it possibly something else?
<img width="802" alt="Screen Shot 2021-05-06 at 12 12 21 PM" src="https://user-images.githubusercontent.com/11756601/117352848-515dda80-ae64-11eb-9694-fedff22710e3.png">
<img width="801" alt="Screen Shot 2021-05-06 at 12 11 34 PM" src="https://user-images.githubusercontent.com/11756601/117352898-5d499c80-ae64-11eb-889f-4cb958990f10.png">

I'm not sure whether there should be repeated eigenvalues with this type of problem. @mlstowell do you know?

Reply:
This problem uses an anisotropic material tensor in the mass matrix. The strongest eigenvalue of this tensor is directed in the (1,1,1) direction, the next in the (0,0,1) direction, and the smallest in the (1,-1,1) direction. This is why the modes tend to be aligned with +/-45 degrees.